weatherman.py

```diff
@@ -1,19 +1,64 @@
+import os
+import glob
 import sys
-sys.argv
 
+sys.argv
 for w in sys.argv:
     print(w)
+if len(sys.argv) < 3:
+    sys.exit(1)
 
 my_weatherlist = []
 
-def func_access(file_name):
-    file = open (file_name , "r+")
+def read_files(file_name):
+    file = open (file_name , "r")
     read = file.read().splitlines()
     for i in range(2, len(read)-1):
-        my_weatherlist.append(read[i].split())
-        print(read[i].split())
-    print(read[0])
+        my_weatherlist.append(read[i].split(","))
+        read[i].split(",")
+
+directory = "./weatherdata"
+year = sys.argv[1]
+sys.argv[3] if len(sys.argv) > 3 and sys.argv[2] == "-e" else None
+
+pattern = os.path.join(directory, year )
```

Comment:
The second argument of the join() will be similar to the one we were using before. The only different will be that instead of providing year in the string like `*2002*.txt` we will be using the year we got from command line argument. For that you have to use string formatting.